src/main/java/com/epam/grid/engine/service/HealthCheckProviderService.java

```diff
@@ -39,43 +31,17 @@
 @Service
 public class HealthCheckProviderService {
 
-    private final EngineType engineType;
-
-    private Map<EngineType, HealthCheckProvider> providers;
-
-    /**
-     * This method sets grid engine type in the context.
-     * @param engineType type of grid engine
-     * @see EngineType
-     */
-    public HealthCheckProviderService(@Value("${grid.engine.type}") final EngineType engineType) {
-        this.engineType = engineType;
-    }
+    @Autowired
+    private HealthCheckProvider healthCheckProvider;
```

Comment:
Let's use constructor-based injection instead of the field one. Add `final` modifier and the required args constructor.
Please, apply that suggestion to other classes

Reply:
done